XmlFormatterOsIndependent/ViewModels/SettingsWindowViewModel.cs

```diff
@@ -0,0 +1,197 @@
+            : base(view, settingsManager, pluginManager)
+        {
+            if (this.settingsManager == null || this.pluginManager == null)
+            {
+                return;
+            }
+            this.settingsManager.Load(settingsPath);
+            applicationScope = this.settingsManager.GetScope("Default");
+            List = pluginManager.ListPlugins<IUpdateStrategy>();
+
+            if (applicationScope == null)
+            {
+                applicationScope = new SettingScope("Default");
+                this.settingsManager.AddScope(applicationScope);
+            }
+            LoadSettings();
+        }
+
+        /// <summary>
+        /// Load all the settings
+        /// </summary>
+        private void LoadSettings()
+        {
+            AskBeforeClosing = GetSettingsValue<bool>("AskBeforeClosing");
+            CheckUpdateOnStart = GetSettingsValue<bool>("SearchUpdateOnStartup");
+            IDataCommand getTheme = new GetThemeCommand();
+            ExecuteCommand(getTheme, settingsManager);
+            ThemeMode = (int)getTheme.GetData<ThemeEnum>();
+
+            string updater = GetSettingsValue<string>("UpdateStrategy");
+            int updaterToUse = -1;
+            if (List.Count > 0)
+            {
+                updaterToUse = 0;
+            }
+
+            for (int i = 0; i < List.Count; i++)
+            {
+                if (List[i].Type.ToString() == updater)
+                {
+                    updaterToUse = i;
+                    break;
+                }
+            }
+
+            if (updaterToUse >= 0)
+            {
+                UpdaterIndex = updaterToUse;
+            }
+        }
+
+        /// <summary>
+        /// Get a specific settings value
+        /// </summary>
+        /// <typeparam name="T">Type of the value to get</typeparam>
+        /// <param name="name">Name of the value to get</param>
+        /// <returns>The value casted to the given type</returns>
+        private T GetSettingsValue<T>(string name)
+        {
+            ISettingPair settingPair = applicationScope.GetSetting(name);
+            return settingPair == null ? default : settingPair.GetValue<T>();
+        }
+
+        /// <summary>
+        /// Close this window
+        /// </summary>
+        public void CloseWindow()
+        {
+            ICommand command = new CloseWindowCommand();
+            ExecuteCommand(command, new CloseWindowData(view.GetWindow()));
+        }
+
+        /// <summary>
+        /// Save the settings and close this window
+        /// </summary>
+        public void SaveAndClose()
+        {
+            ThemeEnum theme = ThemeMode == 0 ? ThemeEnum.Light : ThemeEnum.Dark;
+
+            ISettingPair askClose = new SettingPair("AskBeforeClosing");
+            ISettingPair searchUpdate = new SettingPair("SearchUpdateOnStartup");
+            ISettingPair updater = new SettingPair("UpdateStrategy");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename 'updater' which hides the field with the same name.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5634024)